### PR TITLE
[Improvement-11386][UI] Concise the logic available for task action buttons

### LINKED ISSUE #11386: [Improvement][UI] Concise the logic available for task action buttons on workflow definition detail page.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

It's better to remove the unavailable task action buttons.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sketchmind**
## Purpose of the pull request

Concise the logic available for task action buttons
close #11386 
## Brief change log

* Remove button instead of disable button

* Correct the value of releaseState in the compare statement
## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Review by labbomb — DISMISSED** [review]
LGTM

File: dolphinscheduler-ui/src/views/projects/workflow/components/dag/dag-context-menu.tsx [hunk 1]
```diff
@@ -131,36 +127,37 @@ export default defineComponent({
...
+                {t('project.node.copy')}
+              </NButton>
+              <NButton
+                class={`${styles['menu-item']}`}
+                onClick={this.handleDelete}
+              >
+                {t('project.node.delete')}
+              </NButton>
+            </div>
+          )}
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/workflow/components/dag/dag-context-menu.tsx:159]: The fragment element is better than div here.

File: dolphinscheduler-ui/src/views/projects/workflow/components/dag/index.tsx [hunk 2]
```diff
@@ -119,29 +119,29 @@ export default defineComponent({
...
-          props.instance.state !== 'WAITING_THREAD' &&
-          props.instance.state !== 'SUCCESS' &&
-          props.instance.state !== 'PAUSE' &&
-          props.instance.state !== 'FAILURE' &&
-          props.instance.state !== 'STOP'
+          props.instance.state == 'WAITING_THREAD' ||
+          props.instance.state == 'SUCCESS' ||
+          props.instance.state == 'PAUSE' ||
+          props.instance.state == 'FAILURE' ||
+          props.instance.state == 'STOP'
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/workflow/components/dag/index.tsx]: The strict equality is better here.

File: dolphinscheduler-ui/src/views/projects/workflow/components/dag/dag-context-menu.tsx (hunk 1 above)
Comment (sketchmind) [reply dolphinscheduler-ui/src/views/projects/workflow/components/dag/dag-context-menu.tsx:159]: > The fragment element is better than div here.

Is a '<></>' element ok?

**Review by Amy0104 — APPROVED** [review]
LGTM

**Pull request merged**